.github/workflows/webserver-build-and-push-docker-image.yml

```diff
@@ -0,0 +1,33 @@
+name: Build and push docker image
+on:
+  workflow_call:
+    secrets:
+      dockerhub-username:
+        required: true
+      dockerhub-token:
+        required: true
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Extract commit hash
+        run: echo "GIT_COMMIT_HASH=$(git rev-parse --short "$GITHUB_SHA")" >> "$GITHUB_ENV"
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.dockerhub-username }}
+          password: ${{ secrets.dockerhub-token }}
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./build/webserver/Dockerfile
+          build-args: |
+            GIT_COMMIT_HASH=${{ env.GIT_COMMIT_HASH }}
+            NODE_PORT=3001
+            SERVER_ORIGIN=http://localhost:3001
+            API_BASE_URL=http://user-service:80/v1/users
```

Comment:
The new `webserver` docker image. We briefly considered making this step reusable in [84e9123](https://github.com/Knoblauchpilze/user-service/pull/8/commits/84e9123cf37d426568f2e65a5b088a7671f70863) but the problem was that this would probably have all the environment variables for all the docker images that can be built with this step.

Another approach would be to print the env variables to a file in the calling workflow and then process them here.